airbyte-integrations/connectors/source-github/source_github/streams.py

```diff
@@ -116,25 +116,23 @@ def read_records(self, stream_slice: Mapping[str, any] = None, **kwargs) -> Iter
                     error_msg = (
                         f"Syncing `{self.name}` stream isn't available for repository `{repository}`. Full error message: {error_msg}"
                     )
-            elif e.response.status_code == requests.codes.NOT_FOUND and "/teams?" in error_msg:
-                # For private repositories `Teams` stream is not available and we get "404 Client Error: Not Found for
-                # url: https://api.github.com/orgs/<org_name>/teams?per_page=100" error.
-                error_msg = f"Syncing `Team` stream isn't available for organization `{stream_slice['organization']}`."
-            elif e.response.status_code == requests.codes.NOT_FOUND and "/repos?" in error_msg:
-                # `Repositories` stream is not available for repositories not in an organization.
-                # Handle "404 Client Error: Not Found for url: https://api.github.com/orgs/<org_name>/repos?per_page=100" error.
-                error_msg = f"Syncing `Repositories` stream isn't available for organization `{stream_slice['organization']}`."
-            elif e.response.status_code == requests.codes.GONE and "/projects?" in error_msg:
+            elif e.response.status_code == requests.codes.GONE and isinstance(self, Projects):
                 # Some repos don't have projects enabled and we we get "410 Client Error: Gone for
                 # url: https://api.github.com/repos/xyz/projects?per_page=100" error.
                 error_msg = f"Syncing `Projects` stream isn't available for repository `{stream_slice['repository']}`."
-            elif e.response.status_code == requests.codes.NOT_FOUND and "/orgs/" in error_msg:
+            elif e.response.status_code == requests.codes.NOT_FOUND and isinstance(self, Organizations):
```

Comment:
can't we make this big block much smaller if we have a list of entities which could return 404 and use the same behavior when the 404 is for one of these entities? 

e.g: 
```
OPTIONAL_ENTITIES = [Repositories, Users, Organizations, ...]
....
if 404 and optional_entity
  handle
else: 
  raise
```

Reply:
Improved!